### PR TITLE
sched: Don't duplicate caller file handler when creating kernel thread 

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Fix the regression made by: https://github.com/apache/incubator-nuttx/pull/5379

## Impact
Kernel thread doesn't inherit the caller file handler anymore

## Testing
Local test

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@masayuki2009 could you try this patch? it pass my test locally.

File: sched/group/group_setuptaskfiles.c [hunk 1]
```diff
@@ -57,16 +57,14 @@
...
   FAR struct tcb_s *rtcb = this_task();
   int ret;
 #endif
 
   DEBUGASSERT(group);
-#ifndef CONFIG_DISABLE_PTHREAD
-  DEBUGASSERT((tcb->cmn.flags & TCB_FLAG_TTYPE_MASK) !=
-              TCB_FLAG_TTYPE_PTHREAD);
-#endif
+  DEBUGASSERT(ttype != TCB_FLAG_TTYPE_PTHREAD);
```
Comment (pkarashchenko) [inline sched/group/group_setuptaskfiles.c:67]: Why `#ifndef CONFIG_DISABLE_PTHREAD` is removed?

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
I confirmed that my automatic tests passed.

File: sched/group/group_setuptaskfiles.c (hunk 1 above)
Comment (masayuki2009) [reply sched/group/group_setuptaskfiles.c:67]: @pkarashchenko 
I think the original code is incorrect. The condition should be checked always.

**Pull request merged**